### PR TITLE
Fix nan leak from masked positions in `compute_approx_kl`

### LINKED ISSUE #1633: Bug: `compute_approx_kl` mask-by-multiplication leaks `nan` from masked positions into `policy_kl` / `final_loss`
I had a GRPO step (configured with `apply_overlong_filtering=true`) log `policy_kl: nan` and `final_loss: nan` while `policy_loss: -1.0824`, `grad_norm: 1.5107`, and `policy_entropy: 0.0617` were finite. Looking into this, there's a bug in SkyRL's KL masking in `compute_approx_kl`: https://github.com/NovaSky-AI/SkyRL/blob/skyrl-v0.2.0/skyrl/backends/skyrl_train/utils/ppo_utils.py#L88-L123

The root cause is a one-line issue in `compute_approx_kl` when it masks out non-loss positions with `kld = kld * loss_mask`. Note that `0 * nan = nan`, so any `nan` value at a masked position survives the multiplication, leaks into `masked_mean`'s sum reduction, and poisons every metric that includes the KL scalar.

Notably this is a display-only bug, autograd is unaffected because the gradient value at masked positions is multiplied by `mask=0`, which produces a clean zero regardless of whether the forward value is `nan`.

This reproducer here mirrors what is in `compute_approx_kl` as of SkyRL v0.2.0:

```python
import torch  # 2.10.0
from skyrl.backends.skyrl_train.utils.torch_utils import masked_mean

def compute_approx_kl_buggy(
    log_probs: torch.Tensor,
    log_probs_base: torch.Tensor,
    loss_mask: torch.Tensor,
) -> torch.Tensor:
    """Mirrors SkyRL v0.2.0 compute_approx_kl, k3 branch.

    https://github.com/NovaSky-AI/SkyRL/blob/skyrl-v0.2.0/skyrl/backends/skyrl_train/utils/ppo_utils.py#L88-L123
    """
    kl = log_probs_base - log_probs
    kl = torch.clamp(kl, min=-20, max=20)
    ratio = torch.exp(kl)
    kld = (ratio - kl - 1).contiguous()
    kld = torch.clamp(kld, min=-10, max=10)
    kld = kld * loss_mask                                # bug: 0 * nan == nan
    return kld

def compute_approx_kl_fixed(
    log_probs: torch.Tensor,
    log_probs_base: torch.Tensor,
    loss_mask: torch.Tensor,
) -> torch.Tensor:
    """Same as above but masking via torch.where (proposed fix)."""
    kl = log_probs_base - log_probs
    kl = torch.clamp(kl, min=-20, max=20)
    ratio = torch.exp(kl)
    kld = (ratio - kl - 1).contiguous()
    kld = torch.clamp(kld, min=-10, max=10)
    kld = torch.where(loss_mask.bool(), kld, torch.zeros_like(kld))
    return kld

# Inject nan in log_probs_base at position 1; mask position 1 out of the loss.
log_probs      = torch.tensor([[-1.0, -2.0,         -1.5, -1.7]])
log_probs_base = torch.tensor([[-1.1, float("nan"), -1.4, -1.8]])
loss_mask      = torch.tensor([[ 1.0,  0.0,          1.0,  1.0]])

# Forward: buggy path is nan, fixed path is finite.
kld_b = compute_approx_kl_buggy(log_probs, log_probs_base, loss_mask)
kld_f = compute_approx_kl_fixed(log_probs, log_probs_base, loss_mask)
print("kld (buggy):", kld_b.tolist())
print("kld (fixed):", kld_f.tolist())
print(f"masked_mean (buggy): {masked_mean(kld_b, loss_mask, dim=-1).mean().item()}")
print(f"masked_mean (fixed): {masked_mean(kld_f, loss_mask, dim=-1).mean().item():.6f}")

# Backward: gradients are bit-identical between the two paths.
def kl_b(lp: torch.Tensor) -> torch.Tensor:
    return masked_mean(compute_approx_kl_buggy(lp, log_probs_base, loss_mask), loss_mask, dim=-1).mean()
def kl_f(lp: torch.Tensor) -> torch.Tensor:
    return masked_mean(compute_approx_kl_fixed(lp, log_probs_base, loss_mask), loss_mask, dim=-1).mean()
lp_b = log_probs.clone().requires_grad_(True); (kl_b(lp_b) * 0.01).backward()
lp_f = log_probs.clone().requires_grad_(True); (kl_f(lp_f) * 0.01).backward()
print(f"grad (buggy): {lp_b.grad.tolist()}")
print(f"grad (fixed): {lp_f.grad.tolist()}")
print(f"grad_norm (buggy): {lp_b.grad.norm().item():.6f}")
print(f"grad_norm (fixed): {lp_f.grad.norm().item():.6f}")
```

Output (Python 3.12 on CPU):

```
kld (buggy): [[0.00483739, nan,        0.00517094, 0.00483739]]
kld (fixed): [[0.00483739, 0.0,        0.00517094, 0.00483739]]
masked_mean (buggy): nan
masked_mean (fixed): 0.004949
grad (buggy): [[ 3.17e-04, -0.0, -3.51e-04,  3.17e-04]]
grad (fixed): [[ 3.17e-04, -0.0, -3.51e-04,  3.17e-04]]
grad_norm (buggy): 0.000569
grad_norm (fixed): 0.000569
```

### PR DESCRIPTION
**Pull request by jamesbraza**
Replace `kld * loss_mask` with `torch.where(loss_mask.bool(), kld * loss_mask, 0.0)` in `compute_approx_kl`, so `nan` at masked positions (where `0 * nan = nan`) can no longer leak through into `policy_kl` / `final_loss`.

Closes https://github.com/NovaSky-AI/SkyRL/issues/1633

### CODE REVIEW

File: skyrl/backends/skyrl_train/utils/ppo_utils.py [hunk 1]
```diff
@@ -119,7 +119,9 @@ def compute_approx_kl(
         raise ValueError(f"Invalid KL estimator type: {kl_estimator_type}")
 
     if loss_mask is not None:
-        kld = kld * loss_mask
+        # Multiplying by `loss_mask` can leak `nan` from masked positions,
+        # so fill them with 0.0 instead
+        kld = kld.masked_fill(~loss_mask.bool(), 0.0)
```
[the remark this answers is not in the document]
Comment (jamesbraza) [reply skyrl/backends/skyrl_train/utils/ppo_utils.py]: Oh nice point. I switched to `torch.where(loss_mask.bool(), kld * loss_mask, 0.0)` so soft-mask scaling is preserved while still preventing the nan leak. I also expanded the new unit test to cover this

**Review by erictang000 — APPROVED** [review]
nice, this lgtm, thanks!
